### DISCUSSION TITLE
分享一下满速BT/PT下载的设置方法

### DISCUSSION THREAD

**Discussion by lededev** [Openwrt-Passwall/openwrt-passwall2#103] [Show and tell]
先直接说要点：不要让BT流量经过XRay，否则速度会受影响。
passwall2可以设置流量直接跳过XRay的iptables规则链，只是很多人不太了解其工作原理，所以不会设置罢了。
首先我们需要安排一个直连局域网IP，给BT/PT专用，不要是现有IP，我台式机需要上网，但也用来BT下载，因此需要为它设置两个IP：
192.168.0.111用来上网 **192.168.0.112用来BT下载** 具体怎么设置，稍后再说，还先从passwall2内的设置开始把，先到“**访问控制**”，点击“添加”按钮，新增一条规则
![image](https://user-images.githubusercontent.com/30518126/166325979-6d5aaa99-1914-4361-a07d-1374dc9dea5a.png)

新增详细页填写源地址，可以是多个，并设置TCP不转发端口和UDP不转发端口为：**所有**
![image](https://user-images.githubusercontent.com/30518126/166326200-2d95420c-c778-4d8e-9224-b482212ca4ca.png)
点击保存&应用使之生效。然后转到Windows控制面板的网络设置传统页面：
![image](https://user-images.githubusercontent.com/30518126/166327312-c8db8e4d-d4d5-44e4-99c7-80a8b3b65672.png)
双击IPv4，并点开“高级”对话框，在已有的IP地址111下面，添加一条112：
![image](https://user-images.githubusercontent.com/30518126/166327484-315e9200-570d-4687-a8df-4c55c4db7c15.png)
一路确定使之生效。
最后一步就是设置BT软件了，要让BT下载软件始终使用112这个IP来发包，以便总是直连，不经过XRay。
我用的是qBittorent的Windows版，所以拿它演示如何设置：
![image](https://user-images.githubusercontent.com/30518126/166328048-3640877b-e45e-49c8-acf5-1f0677d98ada.png)
是的，将它绑定到112这个IP，就完成了。之后只要使用这个qBittorent来下载，就是满速，永远不会走代理流量。

**Comment by w316525096** [comment]
请问这样设置的话，bt 的tracker 服务器是不是就连不上了。

**Reply by lededev** [reply]
可以用passwall2开一个socks5代理，bt的tracker全部走那个socks代理

**Comment by hkongm** [comment]
我旁路由自己就在挂 aria2
这种情况咋办啊？passwall 和 aria2 都是同一个ip啊

**Reply by lededev** [reply]
旁路由可以跑pve，openwrt作为其中一个虚拟机跑在旁路由里面，aria2则跑在一个CT容器或者另一个虚拟机里，两边用不同的IP。
这样每次升级openwrt之前可以先做虚拟机快照，升级失败可以用快照恢复到升级前的版本。

**Reply by w316525096** [reply]
openwrt 跑虚拟机里，会不会有性能问题，如果是旁路由还好。

**Reply by hkongm** [reply]
为了这个要弄个 PVE，有点太折腾了，性能倒还好

**Reply by lededev** [reply]
能有什么问题，我的软路由OpenWRT两个10G光纤网口都没性能问题。

**Comment by b9ss** [comment]
我的方法是专门用一台电脑下BT

**Comment by jueduilingdu08** [comment]
QBT能指定IP，但windows好像没办法设置默认哪个IP上网。如果是并行发送的话，有可能导致需要翻墙的数据走了不代理的IP端口发出去，导致翻墙信息泄露真实IP。

**Comment by MadMaxPayneNeo** [comment]
不用这么麻烦，直接开启UPNP和fullcone NAT PT就能跑满带宽
